packages/markdown/parseMarkdown.js

```diff
@@ -238,14 +238,25 @@ function createRender(context) {
         return `<h${level}>${headingHtml}</h${level}>`;
       }
 
-      const headingText = headingHtml
+      let headingText = headingHtml
         .replace(
           /([\uE000-\uF8FF]|\uD83C[\uDC00-\uDFFF]|\uD83D[\uDC00-\uDFFF]|[\u2011-\u26FF]|\uD83E[\uDD10-\uDDFF])\uFE0F?/g,
           '',
         ) // remove emojis
         .replace(/<\/?[^>]+(>|$)/g, '') // remove HTML
```

Comment:
For the history, we added this logic for the support page, e.g. https://mui.com/material-ui/getting-started/support/#blog. I'm not aware of any other places where we do this in the docs.

If we feel that the icons are wrong in the H2, we could also fix it by removing them:

```suggestion
```

Reply:
Regarding emojis, I don't have a strong opinion on this - I'm fine with showing them in ToC or removing them from H2 completely.

Regarding HTML, here is an example H2 from MUI X docs:
```markdown
## Lazy loading [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan)
```
Which is then converted to HTML:
```html
Lazy loading <a href="/x/introduction/licensing/#pro-plan"><span class="plan-pro"></span></a>
```

If we keep the HTML above, React throws an error because of the nested `<a>` tag:
```
Warning: Prop `dangerouslySetInnerHTML` did not match. Server: "Infinite loading " Client: "Infinite loading <a href=\"/x/introduction/licensing/#pro-plan\"><span class=\"plan-pro\"></span></a>
```

This is why I ended up keeping the HTML removal logic and adding the icon later if it's included in `iconsToPersistInToC`. I'm open to a more elegant solution if you have suggestions :)

Reply:
> React throws an error because of the nested <a> tag:

Ah yes, the mismatch seems to come from the nested `<a>`, not good.

```diff
diff --git a/packages/markdown/parseMarkdown.js b/packages/markdown/parseMarkdown.js
index 3d8f9ddafc..3be9e3d23b 100644
--- a/packages/markdown/parseMarkdown.js
+++ b/packages/markdown/parseMarkdown.js
@@ -238,13 +238,8 @@ function createRender(context) {
         return `<h${level}>${headingHtml}</h${level}>`;
       }

-      const headingText = headingHtml
-        .replace(
-          /([\uE000-\uF8FF]|\uD83C[\uDC00-\uDFFF]|\uD83D[\uDC00-\uDFFF]|[\u2011-\u26FF]|\uD83E[\uDD10-\uDDFF])\uFE0F?/g,
-          '',
-        ) // remove emojis
-        .replace(/<\/?[^>]+(>|$)/g, '') // remove HTML
-        .trim();
+      // Remove links to avoid nested links in the TOCs
+      const headingText = headingHtml.replace(/<a\b[^>]*>/i, '').replace(/<\/a>/i, '');

       // Standardizes the hash from the default location (en) to different locations
       // Need english.md file parsed first
```

Reply:
Thanks, this looks better! 🎉

Reply:
Should we keep `<code>` tags in ToC?

| master | This PR |
| ------- | ------- |
| <img width="206" src="https://github.com/mui/material-ui/assets/13808724/96db2ef8-d9f1-428e-b9a9-46473a9ea944"> | <img width="222" src="https://github.com/mui/material-ui/assets/13808724/7bed0330-9820-4d1b-a22d-b52e7a00d632"> |

IMO, it looks slightly better with the `<code>` tags